### ISSUE TITLE
Automatically resolves with arguments even though it wasn't accepted with Tab

### ISSUE THREAD

**Issue by svipas** [zardoy/typescript-vscode-plugins#151]
Even though I don't accept `new URL|` (`|` - cursor) it automatically resolves to `new URL((url))`.

So then I write `new URL(` it becomes this:

```ts
const url = new URL((url))'http://example.com'
```

The question is, why it adds additional `()` around `url` and why it resolves even though I didn't accept it with Tab?

BTW: I tested same happens with `JSON.parse`, etc. it is just enough to add `(` and it automatically resolves.

<img width="215" alt="image" src="https://github.com/zardoy/typescript-vscode-plugins/assets/10441177/38331656-b3bc-4c83-88e1-8c758873dc5b">
<img width="146" alt="image" src="https://github.com/zardoy/typescript-vscode-plugins/assets/10441177/cf7ca6a0-a791-4e0c-a26c-a850ccc61864">

**Comment by svipas** [comment]
Maybe it's related a little bit to: https://github.com/zardoy/typescript-vscode-plugins/issues/142?

**Comment by zardoy** [comment]
See https://github.com/microsoft/vscode/issues/42544

Its also painful to write `...` without shortcut inside objects (https://github.com/microsoft/TypeScript/issues/27623)

**Comment by zardoy** [comment]
> The question is, why it adds additional () around url and why it resolves even though I didn't accept it with Tab?



You can disable it with `"editor.acceptSuggestionOnCommitCharacter": false`

**Comment by zardoy** [comment]
> `const url = new URL((url))'http://example.com'`

However I'm not sure how `'http://example.com'` inserts after `((url))``

Also, if you still prefer to use commit characters I think we can add a workaround for this issue

**Comment by svipas** [comment]
Yeah if I disable it with `"editor.acceptSuggestionOnCommitCharacter": false` it works, but if I disable `methodSnippetsInsertText` it works as expected even with `acceptSuggestionOnCommitCharacter` enabled and it completes function arguments and fixes the problem with for e.g. `new URL(` except that I don't see all arguments in suggest widget.

**Comment by svipas** [comment]
Closing this one since I'm not using `"editor.acceptSuggestionOnCommitCharacter": false` anymore and it works fine for me now. And, I do think it works even with it enabled so probably fixed as well.

**Comment by zardoy** [comment]
afaik it was not fixed

**Comment by svipas** [comment]
But at my first test it seemed to work fine, anyways I'm closing it if you want you can re-open. Is just that I'm not using that setting anymore since it just doesn't work as expected compared to WebStorm.

EDIT: Okay, bug still exists you're right, if you want you can re-open.

**Comment by zardoy** [comment]
yes its now possible to fix with new TS API update
